multimodal-search-pdf/pods/segmenters.py

```diff
@@ -0,0 +1,285 @@
+
+            chunks = []
+            # Extract images
+            with pdf_img:
+                for page in range(len(pdf_img)):
+                    for img in pdf_img.getPageImageList(page):
+                        xref = img[0]
+                        pix = fitz.Pixmap(pdf_img, xref)
+                        # read data from buffer and reshape the array into 3-d format
+                        np_arr = np.frombuffer(pix.samples, dtype=np.uint8).reshape(pix.h, pix.w, pix.n).astype('float32')
+                        if pix.n - pix.alpha < 4:  # if gray or RGB
+                            if pix.n == 1: #convert gray to rgb
+                                np_arr_rgb = np.concatenate((np_arr,)*3,-1)
+                                chunks.append(dict(blob=np_arr_rgb, weight=1.0, mime_type='image/png'))
+                            elif pix.n == 4: # remove transparency layer
+                                np_arr_rgb = np_arr[..., :3]
+                                chunks.append(dict(blob=np_arr_rgb, weight=1.0, mime_type='image/png'))
+                            else:
+                                chunks.append(dict(blob=np_arr, weight=1.0, mime_type='image/png'))
+                        else:  # if CMYK:
+                            pix = fitz.Pixmap(fitz.csRGB, pix)  # Convert to RGB
+                            np_arr = np.frombuffer(pix.samples, dtype=np.uint8).reshape(pix.h, pix.w, pix.n).astype(
+                                'float32')
+                            chunks.append(
+                                dict(blob=np_arr, weight=1.0, mime_type='image/png'))
+
+            # Extract text
+            with pdf_text:
+                count = len(pdf_text.pages)
+                for i in range(count):
+                    page = pdf_text.pages[i]
+                    text_page = page.extract_text(x_tolerance=1, y_tolerance=1)
+                    if text_page:
+                        chunks.append(dict(text=text_page, weight=1.0, mime_type='text/plain'))
+        return chunks
+
+
+class TextSegmenter(BaseSegmenter):
+    """
+    :class:`TextSegmenter` simply store text into chunks.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @single(slice_nargs=4)
+    def segment(self, text: str, uri: str, buffer: bytes, mime_type: str, *args, **kwargs) -> List[Dict]:
+        """
+        Segements text.
+
+        :param text: the text data
+        :param uri: File name of PDF
+        :type uri: str
+        :param buffer: PDF file in bytes
+        :type buffer: bytes
+        :param mime_type: the type of data
+        :returns: A list of documents with the extracted data
+        :rtype: List[Dict]
+        """
+
+        chunks = []
+        if mime_type == 'text/plain':
+            chunks.append(dict(text=text, weight=1.0, mime_type='text/plain'))
+        return chunks
+
+
+class TextSegmenterCustomized(BaseSegmenter):
```

Comment:
Make it clear that it is inherited from `Segmenter`
```suggestion
class CustomizedTextSegmenter(BaseSegmenter):
```